libkbfs/tlf_journal.go

```diff
@@ -411,67 +428,57 @@ func (j *tlfJournal) flush(ctx context.Context) (err error) {
 		}
 	}()
 
-	// TODO: Interleave block flushes with their related MD
-	// flushes.
+	// TODO: Avoid starving flushing MD ops if there are many
+	// block ops. See KBFS-1502.
 
-	// Flush the block journal ops in parallel.
 	for {
-		numFlushed, err := j.flushBlockEntries(ctx)
+		blockEnd, mdEnd, err := j.getJournalEnds(ctx)
 		if err != nil {
 			return err
 		}
-		if numFlushed == 0 {
+
+		j.log.CDebugf(ctx, "Flushing up to blockEnd=%d and mdEnd=%d",
+			blockEnd, mdEnd)
+
+		if blockEnd == 0 && mdEnd == MetadataRevisionUninitialized {
 			break
 		}
-		flushedBlockEntries += numFlushed
-	}
 
-	// TODO: flush MDs in parallel?
```

Comment:
We'd have to add a new RPC to the server, but it should be possible.  Maybe "batch" is a better word?  Would be nice to keep the TODO though.